charts/prometheus-operator.yaml

```diff
@@ -38,6 +38,19 @@ alertmanager:
             matchLabels:
               app.kubernetes.io/name: prometheus-operator-alertmanager
 
+  ## Passing the below configuration directives through the helm template
+  ## engine leads to no changes.
+  ## So let us create a new secret that overrides/patches
+  ## alertmanager-prometheus-operator-alertmanager secret with new config
+  ## read from a ConfigMap (metalk8s-alertmanager-config)
+  ## This implies we don't have to worry about patching the local charts
```

Comment:
Do we need to patch lot of stuff in the chart ? Because if for what ever reason the secret chart change in another version we will not get the changes

Reply:
I think the main issue here was that each variable passed was directly base64 encoded. This is the current behaviour of the chart.

If we rationally go by the way we write, substitute and dump the jinja variables, this meant we will have something like `global: __var_tojson__(alertmanager.spec.notification.config.global)` inside `alertmanager.yaml` once we base64 decode.

Reply:
Ok make sense, let keep your approach then